site/app/libraries/database/DatabaseQueries.php

```diff
@@ -4146,6 +4146,44 @@ public function removeFromSeekingTeam($g_id, $user_id) {
         $this->course_db->query("DELETE FROM seeking_team WHERE g_id=? AND user_id=?", [$g_id, $user_id]);
     }
 
+    /**
+     * Return an array of Team objects(team_id, team_name, team_members) for all teams on given gradeable
+     *
+     * @param Gradeable $gradeable
+     * @return \app\models\Team[]
+     */
+    public function getTeamsByGradeable(Gradeable $gradeable) {
+        $this->course_db->query(
+            "
+            SELECT 
+                gt.team_id AS team_id,
+                gt.team_name AS team_name,
+                STRING_AGG(u.user_id, ',') AS team_members
+            FROM 
+                gradeable_teams gt
+            JOIN 
+                teams t ON gt.team_id = t.team_id
+            JOIN 
+                users u ON t.user_id = u.user_id
+            WHERE 
+                gt.g_id = ?
+            GROUP BY 
+                gt.team_id
+            ORDER BY
+                team_id",
+            [$gradeable->getId()]
+        );
+        $teams = [];
+        $rows = $this->course_db->rows();
+        if ($rows !== null) {
```

Comment:
rows will always be of type array (never a null case), so the only thing it could be is an empty array. Therefore I don't think you need this check

Reply:
You are right 
it worked well when I checked the size of array instead of checking null

Reply:
Thanks for the suggestions @04vmatsibekker 
Most of them worked correctly

and right when I pushed the changes, new php static errors popped up
if its this way, I think there will be lot more lint/static errors 
May be I should let one of the maintainers know about this issue ?